src/test/functions/projectConfig.spec.ts

```diff
@@ -120,4 +120,26 @@ describe("projectConfig", () => {
       ).to.throw(FirebaseError, /functions.codebase must be unique/);
     });
   });
+
+  describe("suggestCodebaseName", () => {
+    it("changes uppercase characters to lowercase", () => {
+      expect(projectConfig.suggestCodebaseName("COdeBaSE")).to.equal("codebase");
+    });
+
+    it("replaces invalid characters with underscores", () => {
+      expect(projectConfig.suggestCodebaseName("c!o@d#e$b%a^s&e*")).to.equal("c_o_d_e_b_a_s_e_");
+    });
+
+    it("truncates names that are too long", () => {
+      expect(
+        projectConfig.suggestCodebaseName(
+          "1234567812345678123456781234567812345678123456781234567812345678"
+        )
+      ).to.equal("123456781234567812345678123456781234567812345678123456781234567");
+    });
+
+    it("suggests valid name based on name with multiple issues", () => {
+      expect(projectConfig.suggestCodebaseName("Codebase#1")).to.equal("codebase_1");
```

Comment:
nit* I might also assert that suggested codebase is also a valid codebase by running it through `validateCodebase`

Reply:
so after feedback from the API council I don't actually use the`suggestCodebaseName` utility lol. Should I keep the function in `projectConfig.ts` and this corresponding unit test in case it is useful in the future, or remove it for now?